src/stan/model/indexing/rvalue.hpp

```diff
@@ -144,25 +144,31 @@ inline auto rvalue(Vec&& v, const char* name, index_uni idx) {
  * Types:  vector[multi] = vector
  *
  * @tparam EigVec Eigen type with either dynamic rows or columns, but not both.
+ * @tparam MultiIndex A multi index.
  * @param[in] v Eigen vector type.
  * @param[in] name Name of variable
  * @param[in] idx Sequence of integers.
  * @throw std::out_of_range If any of the indices are out of bounds.
  * @throw std::invalid_argument If the value size isn't the same as
  * the indexed size.
  */
-template <typename EigVec, require_eigen_vector_t<EigVec>* = nullptr>
-inline auto rvalue(EigVec&& v, const char* name, const index_multi& idx) {
+template <typename EigVec, typename MultiIndex,
+          require_eigen_vector_t<EigVec>* = nullptr,
+          require_same_t<MultiIndex, index_multi>* = nullptr>
```

Comment:
What is the purpose of the template if you're constraining it to one concrete type?

Reply:
Perfect forwarding